### ISSUE TITLE
IntelliSense for config

### ISSUE THREAD

**Issue by michaeloliverx** [pngwn/MDsveX#300]
Packages that ship with TypeScript typings can leverage IDE's intellisense with jsdoc type hints in JS files.

[Vite](https://vitejs.dev/config/#config-intellisense)
```js
/**
 * @type {import('vite').UserConfig}
 */
const config = {
  // ...
}

export default config
```

SvelteKit
```js
/**
 * @type {import('@sveltejs/kit').Config}
 */
const config = {
	// ...
};

export default config;
```

[Tailwind](https://github.com/tailwindlabs/tailwindcss/discussions/1077#discussioncomment-528222)
```js
/** @type {import("@types/tailwindcss/tailwind-config").TailwindConfig } */
```

Unfortunately I cannot get this working with MDsveX:
```js
/**
 * @type {import('mdsvex').MdsvexOptions}
 **/
const config = {
	extensions: ['.svx', '.md']
};

module.exports = config;
```

`config` is still `any`

I noticed when typing out the path for the others the types/ts files are listed:
<img width="522" alt="image" src="https://user-images.githubusercontent.com/55017335/125201564-a3a5e800-e267-11eb-96b5-e565410a703b.png">

But when doing the same for mdsvex no typescript files are shown:
<img width="462" alt="image" src="https://user-images.githubusercontent.com/55017335/125201612-cdf7a580-e267-11eb-8218-b96ba9ecfa87.png">

Perhaps its an issue how the types are being bundled? (I have no idea how the npm packaging works but just a guess)

--- 

Its not majorly important but it is _nice_ 😄

_Originally posted by @michael0liver in https://github.com/pngwn/MDsveX/discussions/280_

**Comment by michael0liver** [comment]
As a workaround you can use the [`Parameters<Type>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype)  utility type to get IntelliSense for the config object.
```js
// mdsvex.config.js
/** @type {Parameters<typeof import("mdsvex").mdsvex>[0]} */
export const mdsvexConfig = {
  ...
}
```

Ideally we would be able to do the following: `/** @type {import("mdsvex").MdsvexOptions} */`

**Comment by pngwn** [comment]
V1 is being worked and will support both:

```ts
/**
 * @type {import('mdsvex').MdsvexOptions}
 **/
 export const mdsvex_config = { ... };
```

And a config function that provides types:

```ts
import { config } from 'mdsvex';

export const mdsvex_config = config({ ... });
```

This is similar to how `vite` works.

**Comment by pngwn** [comment]
I also have no idea why it isn't working but is probably a simple thing.

**Comment by michael0liver** [comment]
Some additional information that might help:

The source exports the `MdsvexOptions` interface:

https://github.com/pngwn/MDsveX/blob/7bd9885b4d01d99c9f38e10d4cd4a931c14ffbb6/packages/mdsvex/src/types.ts#L198

The generated `.d.ts` files don't reflect this:
<img width="540" alt="image" src="https://user-images.githubusercontent.com/55017335/132872793-a45444b0-1916-4838-b6d1-54a4ee7ec4ac.png">

I feel like it could be a tooling issue, 

https://github.com/pngwn/MDsveX/blob/7bd9885b4d01d99c9f38e10d4cd4a931c14ffbb6/packages/mdsvex/package.json#L11

Is `dts-generator` needed? I think you can just use the typescript compiler to emit declaration files, maybe take a look at the `declaration` https://www.typescriptlang.org/tsconfig/#declaration option.

**Comment by pngwn** [comment]
I'm not actually using it, it is probably just a leftover from my struggles to generate types the way I wanted to. I'm using `rollup-plugin-dts` to generate type iirc. This is because I want to bundle types together as much as possible and not require that a bunch of transitory dependencies have to be installed just for types. The TS compiler does not support this out of the box.

**Comment by PuruVJ** [comment]
Hey folks just added a `defineMDSvexConfig` option. Open to bike shedding

**Comment by pngwn** [comment]
This has been release in version 0.10.2 thanks @PuruVJ! Would love any feedback.

**Comment by pngwn** [comment]
Had to revert this and re-release, everything broke.

**Comment by pngwn** [comment]
Resolved in #409 released in `0.10.5`
